records/labels/spec.go

```diff
@@ -0,0 +1,58 @@
+package labels
+
+import (
+	"fmt"
+)
+
+const (
+	// SeparatorChar delimits concatenated labels.
+	SeparatorChar = '.'
+
+	DNS952MaxLength  int = 24
+	DNS1123MaxLength int = 63
+
+	// see http://www.rfc-base.org/txt/rfc-952.txt
+	HostNameSpecType952 HostNameSpecType = iota
+
+	// see http://www.rfc-base.org/txt/rfc-1123.txt
+	HostNameSpecType1123
```

Comment:
I tend to make exceptions for Go in this area when it comes to error
types/implementations. Otherwise, I think the package should probably guard
against clients (possibly unintentionally) stomping on the original values.


On Fri, Jun 26, 2015 at 1:39 AM, Tomás Senart <notifications@github.com>
wrote:

> In records/labels/spec.go
> <https://github.com/mesosphere/mesos-dns/pull/177#discussion_r33329853>:
>
> > +import (
> > +	"fmt"
> > +)
> > +
> > +const (
> > +	// SeparatorChar delimits concatenated labels.
> > +	SeparatorChar = '.'
> > +
> > +	DNS952MaxLength  int = 24
> > +	DNS1123MaxLength int = 63
> > +
> > +	// see http://www.rfc-base.org/txt/rfc-952.txt
> > +	HostNameSpecType952 HostNameSpecType = iota
> > +
> > +	// see http://www.rfc-base.org/txt/rfc-1123.txt
> > +	HostNameSpecType1123
>
> I usually just accept this as a limitation of Go not having immutable
> variables. I don't think it's necessary to consider the risk of a package
> variable being overwritten by users as that is not its intended use.
> … <#14e2e612357bd5e8_>
> On Thu 25 Jun 2015 at 22:32 James DeFelice <notifications@github.com>
> wrote: In records/labels/spec.go <
> https://github.com/mesosphere/mesos-dns/pull/177#discussion_r33329669>: >
> +import ( > + "fmt" > +) > + > +const ( > + // SeparatorChar delimits
> concatenated labels. > + SeparatorChar = '.' > + > + DNS952MaxLength int =
> 24 > + DNS1123MaxLength int = 63 > + > + // see
> http://www.rfc-base.org/txt/rfc-952.txt > + HostNameSpecType952
> HostNameSpecType = iota > + > + // see
> http://www.rfc-base.org/txt/rfc-1123.txt > + HostNameSpecType1123 I
> remember my initial reasoning for doing it this way: I didn't want to
> create globals that were non-constant but that should have constant,
> deterministic behavior. Upon looking at my interfaces I think they still
> leak a bit, still allowing for the original structs to be modified by code
> outside of the package (see http://play.golang.org/p/vVInOnjtt0). I'll
> stop leaking the pointers to the original structs. …
> <#msg-f:1505017743443090707_> On Thu, Jun 25, 2015 at 11:51 PM, James
> DeFelice <james@mesosphere.io> wrote: > Yep, probably. I'll keep the
> constants for length though - trying to avoid > introducing magic numbers
> not assigned to constants. > > On Thu, Jun 25, 2015 at 10:14 PM, Tomás
> Senart <notifications@github.com> > wrote: > In records/labels/spec.go >
> < https://github.com/mesosphere/mesos-dns/pull/177#discussion_r33324053>:
> > > > +import ( > > + "fmt" > > +) > > + > > +const ( > > + //
> SeparatorChar delimits concatenated labels. > > + SeparatorChar = '.' > > +
> > > + DNS952MaxLength int = 24 > > + DNS1123MaxLength int = 63 > > + > > +
> // see http://www.rfc-base.org/txt/rfc-952.txt > > + HostNameSpecType952
> HostNameSpecType = iota > > + > > + // see
> http://www.rfc-base.org/txt/rfc-1123.txt > > + HostNameSpecType1123 > > I
> believe this can be further simplified. No need for HostNameSpecType. > >
> var ( > RFC952 = HostNameSpec{table: makeDNS952Table(), maxlen: 24} >
> RFC1123 = HostNameSpec{table: makeDNS1123Table(), maxlen: 63} > ) > > — >
> Reply to this email directly or view it on GitHub > <
> https://github.com/mesosphere/mesos-dns/pull/177/files#r33324053>. > —
> Reply to this email directly or view it on GitHub <
> https://github.com/mesosphere/mesos-dns/pull/177/files#r33329669>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/177/files#r33329853>.
>